app/js/patternFunc.js

```diff
@@ -0,0 +1,63 @@
+/*exported patternF */
+
+function patternF() {
+
+  "use strict";
+
+  function plant(arg) {
+    var that = {};
```

Comment:
Something is wrong here. 
that is often used for this in closures. Here it could be plantOptions or whatever